staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -219,10 +221,42 @@ func (az *Cloud) EnsureLoadBalancerDeleted(ctx context.Context, clusterName stri
 
 // GetLoadBalancerName returns the LoadBalancer name.
 func (az *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	// TODO: replace DefaultLoadBalancerName to generate more meaningful loadbalancer names.
+	if lbName, found := service.Annotations[LoadBalancerName]; found && strings.TrimSpace(lbName) != "" {
+		return lbName
+	}
 	return cloudprovider.DefaultLoadBalancerName(service)
 }
 
+func (az *Cloud) getLoadBalancerResourceGroup(service *v1.Service) string {
+	if rgName, found := service.Annotations[ServiceAnnotationLoadBalancerResourceGroup]; found && strings.TrimSpace(rgName) != "" {
```

Comment:
let's add two new configuration options in `Config`:

```go
LoadBalancerName string `json:"loadBalancerName,omitempty" yaml:"loadBalancerName,omitempty"`
LoadBalancerResourceGroup string `json:"loadBalancerResourceGroup,omitempty" yaml:"loadBalancerResourceGroup,omitempty"`
```

Reply:
So, `getLoadBalancerResourceGroup` could be 

```go
if az.LoadBalancerResourceGroup != "" {
  return az.LoadBalancerResourceGroup
}

return az.ResourceGroup
```

Reply:
same for loadBalancerName. But notice `az.LoadBalancerName` is only applied for defaultLBName